content/en/docs/tasks/administer-cluster/access-cluster-api.md

```diff
@@ -219,12 +220,29 @@ The easiest way to use the Kubernetes API from a Pod is to use
 one of the official [client libraries](/docs/reference/using-api/client-libraries/). These
 libraries can automatically discover the API server and authenticate.
 
+#### Using Official Client Libraries
+
+From within a Pod, the recommended ways to connect to the Kubernetes API are:
+
+  - For a Go client, use the official [Go client library](https://github.com/kubernetes/client-go/).
+    The `rest.InClusterConfig()` function handles API host discovery and authentication automatically.
+    See [an example here](https://git.k8s.io/client-go/examples/in-cluster-client-configuration/main.go).
+
+  - For a Python client, use the official [Python client library](https://github.com/kubernetes-client/python/).
+    The `config.load_incluster_config()` function handles API host discovery and authentication automatically.
+    See [an example here](https://github.com/kubernetes-client/python/blob/master/examples/in_cluster_config.py).
+
+In each case, the service account credentials of the Pod are used to communicate
+securely with the API server.
+
+#### Directly accessing the REST API
+
 While running in a Pod, the Kubernetes apiserver is accessible via a Service named
 `kubernetes` in the `default` namespace. Therefore, Pods can use the 
 `kubernetes.default.svc` hostname to query the API server. Official client libraries
 do this automatically.
 
-From within a Pod, the recommended way to authenticate to the API server is with a
+The recommended way to authenticate to the API server is with a
```

Comment:
From within a Pod, clients should always authenticate using service account credentials. From outside the cluster? Maybe the client represents a cluster operator (the human kind), who should use whatever user-authenticating mechanism(s) that cluster offers.

For that reason I like the original text better.

Reply:
From outside the cluster, is documented in a completely separate section, further up the page.